Tracks/neftovsky_Floating solo toolbar.lua

```diff
@@ -0,0 +1,419 @@
+      reaper.ImGui_PushStyleVar(ctx, reaper.ImGui_StyleVar_FramePadding(), 8 * visual.sizeX, 8 * visual.sizeY)
+
+      if reaper.ImGui_Button(ctx, b.button_label .. "##" .. key) then
+        local alt = reaper.ImGui_GetKeyMods(ctx) & reaper.ImGui_Mod_Alt() ~= 0
+        local ctrl = reaper.ImGui_GetKeyMods(ctx) & reaper.ImGui_Mod_Ctrl() ~= 0
+
+        if ctrl and visual.enableCtrlMute then ToggleMute(names, b.partialNameMatch)
+        elseif alt then SoloExclusive(key)
+        else ToggleSolo(names, b.partialNameMatch) end
+      end
+
+      if reaper.ImGui_IsItemClicked(ctx, 1) then
+        selected_button_key = key
+        button_name_buffer = b.button_label
+        track_name_buffer = b.track_name
+        show_visual_menu = false
+        reaper.ImGui_OpenPopup(ctx, "Edit")
+      end
+
+      reaper.ImGui_PopStyleVar(ctx, 2)
+      reaper.ImGui_PopStyleColor(ctx, 2)
+
+      if not visual.verticalLayout then
+        reaper.ImGui_SameLine(ctx, nil, visual.spacingX)
+      end
+    end
+
+ if reaper.ImGui_BeginPopup(ctx, "Edit") then
+     _, button_name_buffer = reaper.ImGui_InputText(ctx, "Label", button_name_buffer, 256)
+     _, track_name_buffer = reaper.ImGui_InputText(ctx, "Track", track_name_buffer, 256)
+ 
+     -- Кнопки OK и Cancel с галочкой в одной строке
+     if reaper.ImGui_Button(ctx, "OK") then
+         if selected_button_key and buttons[selected_button_key] then
+             buttons[selected_button_key].button_label = button_name_buffer
+             buttons[selected_button_key].track_name = track_name_buffer
+             SaveSettings()
+         end
+         reaper.ImGui_CloseCurrentPopup(ctx)
+     end
+     reaper.ImGui_SameLine(ctx)
+     if reaper.ImGui_Button(ctx, "Cancel") then reaper.ImGui_CloseCurrentPopup(ctx) end
+     reaper.ImGui_SameLine(ctx)
+     if selected_button_key and buttons[selected_button_key] then
+         _, buttons[selected_button_key].partialNameMatch = reaper.ImGui_Checkbox(ctx, "Contains in name", buttons[selected_button_key].partialNameMatch)
+     end
+ 
+     reaper.ImGui_Separator(ctx)
+     if reaper.ImGui_Button(ctx, "+") then
+         local new_key = "Button" .. tostring(#button_order + 1)
+         buttons[new_key] = {track_name = "New Track", button_label = "New", partialNameMatch = true}
+         table.insert(button_order, new_key)
+         SaveSettings()
+     end
+     reaper.ImGui_SameLine(ctx)
+     if reaper.ImGui_Button(ctx, "-") then
+         local last = table.remove(button_order)
+         if last then buttons[last] = nil end
+         SaveSettings()
+     end
+ 
+     reaper.ImGui_Separator(ctx)
+     if reaper.ImGui_Button(ctx, "Settings") then show_visual_menu = not show_visual_menu end
+     if show_visual_menu then
+         reaper.ImGui_Separator(ctx)
+         RenderVisualSettings()
+     end
+ 
+     reaper.ImGui_EndPopup(ctx)
+ end
+
+    reaper.ImGui_End(ctx)
+  end
+
+  if open then reaper.defer(Main) end
+end
+
+Main()
```

Comment:
To fix the first frame having double the contents:
```suggestion
reaper.defer(Main)
```